### PR TITLE
Improve the handling of auto battle on/off events and log them properly

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
The idea is to throw away that hacky approach with `Arena::BreakAutoBattle()` and use the "regular" approach with `Battle::Command`. Perhaps not the ideal approach, but I think it's less "hacky" and more "general" than the current one.

### CODE REVIEW

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @oleg-derevenetz , I put few comments in this pull request. Could you please check them if they make sense?

File: src/fheroes2/battle/battle_arena.cpp [hunk 1]
```diff
@@ -324,7 +324,15 @@ void Battle::Arena::TurnTroop( Unit * troop, const Units & orderHistory )
     while ( !end_turn ) {
         Actions actions;
 
-        if ( !troop->isValid() ) { // looks like the unit died
+        if ( interface ) {
+            interface->getPendingActions( actions );
+        }
+
+        if ( !actions.empty() ) {
```
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/battle/battle_arena.cpp:331]: @ihhub `MSG_BATTLE_AUTO` is already ignored for the purpose of seed generation. And other possible future pure GUI-related actions should be ignored too. There is nothing new in this PR in this regard.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/battle/battle_arena.cpp:331]: Thank you very much for the explanation!

**Comment by ihhub** [comment]
@oleg-derevenetz , thanks a lot for improving the code!

**Pull request merged**